mlflow/keras/autologging.py

```diff
@@ -224,10 +198,34 @@ class FitPatch(PatchFunction):
         def __init__(self):
             pass
 
+        def _infer_batch_size(self, inst, *args, **kwargs):
+            batch_size = None
+            if "batch_size" in kwargs:
+                batch_size = kwargs["batch_size"]
+            else:
+                training_data = kwargs["x"] if "x" in kwargs else args[0]
+                if _batch_size := getattr(training_data, "batch_size", None):
+                    batch_size = _batch_size
+                elif _batch_size := getattr(training_data, "_batch_size", None):
+                    batch_size = (
+                        _batch_size if isinstance(_batch_size, int) else _batch_size.numpy()
+                    )
+                elif is_iterator(training_data):
+                    is_single_input_model = isinstance(inst.input_shape, tuple)
+                    training_data, training_data_copy = tee(training_data)
```

Comment:
Is this the main change?

Reply:
Yes

Reply:
The test passes here: https://github.com/mlflow/mlflow/actions/runs/8059394201/job/22013742169